pkg/build/templates/modify-raw-image.sh.tpl

```diff
@@ -13,6 +13,13 @@ set -euo pipefail
 #
 # Guestfish Command Documentation: https://libguestfs.org/guestfish.1.html
 
+# Test the block size of the base image and adapt to suit either 512/4096 byte images
+BLOCKSIZE=512
+if ! guestfish -i --blocksize=$BLOCKSIZE -a {{.ImagePath}} echo "[INFO] 512 byte sector check successful."; then
+        echo "[WARN] Failed to access image with 512 byte sector size, trying 4096 bytes."
+        BLOCKSIZE=4096
```

Comment:
Shall we do the same check here instead of assuming 4096?

Reply:
@e-minguez thanks for checking this, but this statement *is* the check. We start with assuming 512 bytes, we then run `guestfish` with a 512 byte sector size, and if that fails (it will show up as `operating system not found` in the logs), we throw a warning and assume 4096. This value then feeds into the next call in the script, the full `guestfish` run where we run the necessary modifications on the raw disk image.

Reply:
what I was saying is something like:

```
BLOCKSIZE=512
if ! guestfish -i --blocksize=$BLOCKSIZE -a {{.ImagePath}} echo "[INFO] 512 byte sector check successful."; then
  echo "[WARN] Failed to access image with 512 byte sector size, trying 4096 bytes."
  BLOCKSIZE=4096
  if ! guestfish -i --blocksize=$BLOCKSIZE -a {{.ImagePath}} echo "[INFO] 4096 byte sector check successful."; then
    echo "[ERROR] Failed to access image with 512 or 4096 byte sector sizes."
fi
```

(this can be improved of course)

Reply:
Are there other blocksizes we need to accomodate for? 

Is there any way the check could return a false positive?

Reply:
not that I'm aware of, but _just in case_ I guess it won't hurt to check both :)

Reply:
I hear what you're saying, Edu, but each `guestfish` call adds to the image build time. I think we should assume 512 (as the vast majority of situations will be 512) and if that fails, try with 4096 (the only other somewhat common block size). I don't think it makes sense to check for 4096 before proceeding, at this point we should just error out.

Reply:
Agree!